src/components/Player.js

```diff
@@ -135,16 +142,28 @@ class SanityMuxPlayer extends Component {
   }
 
   handleVideoClick = event => {
-    this.setState({showControls: true})
-    if (this.hls) {
-      this.hls.startLoad(0)
+    if (!autoload) {
+      this.setState({showControls: true})
+      if (this.hls) {
+        this.hls.startLoad(0)
+      }
+    }
+    if (this.props.onClick) {
+      this.props.onClick(event)
     }
   }
 
   // eslint-disable-next-line complexity
   render() {
     const {posterUrl, isLoading, error} = this.state
-    const {assetDocument, autoload} = this.props
+    const {
+      assetDocument, autoload,
+      showControls: showControlsProp,
```

Comment:
What is `showControlsProp`?

Reply:
Further down, on lines 181-184 `showControls` is introduced with `let` and get special treatment.

In this block, though, I destructure all props that are getting handled by `SanityMuxPlayer` internally and spread the unhandled ones into `videoProps` for safe forwarding.

Since `showControls` is the name of the prop but we also introduce `showControls` on L181, upon destructuring it here, I rename it to `showControlsProp` such that it doesn't collide with the `let showControls` call later.

Reply:
If this is too hard to follow, and I see it is and I feared it would be, probably we should go another route for filtering unhandled props.

Introducing  `omit`, for instance.  
What I like about destructure + spread, though, is that it is actually ES6's native way of `omit`.

P.S. While writing this, I noticed that I am not lifting `onClick` out, so if `onClick` was provided to `SanityMuxPlayer`, it'd override the internal handler assignment. Meaning, in current scenario, `onClick` should also get assigned to a variable that's not actually getting used in this scope... Massive pollution, more incentive to go with `omit` utility.